tt_metal/hw/ckernels/blackhole/metal/llk_api/llk_sfpu/llk_math_eltwise_unary_sfpu_mask.h

```diff
@@ -18,11 +18,11 @@ inline void llk_math_eltwise_unary_sfpu_mask_init() {
 }
 
 template <bool APPROXIMATE>
-inline void llk_math_eltwise_unary_sfpu_mask(uint dst_index, int vector_mode = (int)VectorMode::RC) {
-    llk_math_eltwise_unary_sfpu_params<APPROXIMATE>(
-        ckernel::sfpu::calculate_mask<APPROXIMATE>,
-        dst_index,
-        vector_mode);
+inline void llk_math_eltwise_unary_sfpu_mask(uint dst_index, DataFormat data_format, int vector_mode = (int)VectorMode::RC) {
+    if (data_format == DataFormat::Float16_b || data_format == DataFormat::Float16) {
+        llk_math_eltwise_unary_sfpu_params<APPROXIMATE>(
+            ckernel::sfpu::calculate_mask<APPROXIMATE>, dst_index, vector_mode);
+    }
```

Comment:
Should there be an assert or a message for when the dataformats are not Float16/Float16b? What happens if it is other formats?

Reply:
> What happens if it is other formats?
If other Data Format parameters are passed, mask_tile will not function.

> Should there be an assert or a message for when the dataformats are not Float16/Float16b? 
I tried to add runtime asserts or error messages in LLK but couldn't find any examples other than static_assert. If there is a method to implement runtime asserts or messages in LLK, could you please let me know?

Reply:
LLKs unfortunately dont have runtime asserts, you can place asserts in the higher level, for example op_library:
`tt_eager/tt_dnn/op_library/moreh_sum/moreh_sum_nc_impl/moreh_int_sum_nc_impl.cpp` in case incorrect data formats are used?

Reply:
There is an assert section in moreh_sum_op.cpp before the *_impl.cpp file, specifically at [lines 182-183](https://github.com/tenstorrent/tt-metal/blob/07a51050c700f468e62e2e8fd45a57f73dfc376d/tt_eager/tt_dnn/op_library/moreh_sum/moreh_sum_op.cpp#L182-L183), to handle incorrect data formats. :)

Reply:
ok awesome that works!